### PR TITLE
Jest testing part1

### PR DESCRIPTION
**Pull request by Alzubair98**
all the test requirements are done
its ready for lunch.

### CODE REVIEW

File: __test__/add_remove.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { addElement, deletetask } from '../src/add_remove.js';
```
Comment (vechicin) [inline __test__/add_remove.test.js:1]: Please make sure you test most, if not all, functions involved in adding or deleting tasks. You only tested one for each.

Screen Shot 2022-07-13 at 18 44 14

**Review by vechicin — CHANGES_REQUESTED** [review]
# Hi @Alzubair98 ✋, 

Good job so far! 🔥 
There are some issues that you still need to work on to go to the next project but you are almost there! 🚨 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights ⚡ 

- [x] All tests are passing!

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions 📂 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: __test__/add_remove.test.js (hunk 1 above)
Comment (for-ashraf) [reply __test__/add_remove.test.js:1]: Dear, look at the requirement No.4, i.e.: 
**Take a look at each of these functions. Are they pure functions? If the answer is "yes" then writing tests for them should be straightforward.**
So, the addition and deletion are the only pure functions, functions related to these functions are not pure?

File: __test__/add_remove.test.js [hunk 2]
```diff
@@ -1,16 +1,21 @@
-import { addElement, deleteElement } from '../src/add_remove.js';
+import { addElement, deletetask } from '../src/add_remove.js';
```
Comment (Kingstalux) [inline __test__/add_remove.test.js:1]: - [ ] Good job so far 👍 but as the previous reviewer mentioned you have to test almost everything related to adding or removing a task and not only pure functions. You also have to mock the local storage and write tests to make sure the local storage is updated accordingly. You also have to mock HTML to test if add/delete functions add or remove exactly one `li` element to/from the list in the DOM as stated clearly in the requirements.
- Requirements:
Screenshot 2022-07-14 092131

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
175083599-879bf673-fb93-406f-939c-de7e586bf6d9


 ## Highlights
✅ Good commit history 👍 
✅ Linters are passing 👌 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: __test__/add_remove.test.js (hunk 2 above)
Comment (Alzubair98) [reply __test__/add_remove.test.js:1]: https://www.loom.com/share/3f1a624abf2d4faf904d5f6c94b164f8

**Comment by Kingstalux** [comment]
Hello @Alzubair98, Good job adding more tests, especially those mocking the local storage 👍 you can submit your project for approval now 😄

**Review by prolajumokeoni — APPROVED** [review]
Hi @Team 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Github flow was used correctly 🚀 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**